gitgud/levels/rampup/_reversing/instructions.txt

```diff
@@ -1,19 +1,17 @@
 >>>
-The third option is the most extreme (but may be the most useful). When you want to rollback the clock completely to a certain commit: use "git reset --hard <commit_ref>". This command will annihilate any changes you've made since the specified commit. Essentially, using this command will reset your repository to exactly what it was at the time of the specified commit. Be careful with this option!
+The second method is "git revert". Using "git revert" will create a new commit that will undo the changes that were made in a specified commit. For example, suppose I commit three files, 1, 2, and 3(<-HEAD). Using "git revert HEAD" will create a fourth commit where there are only files 1 and 2. In other words, the changes made at HEAD (commit 3) will be undone. Similarly, I could have used "git revert HEAD^" and this would have created a fourth commit with only files 1 and 3.
 >>>
-I mentioned "git reset" isn't the only revision command.
+Notice that in the previous example the commit with files 1, 2, and 3 still exists in the commit history! Git revert allows you to keep track of the commits prior to the revert. Therefore, if we decided we actually needed those changes, we can return to them. If you recall, this is unlike reset which abandons the working commit.
 >>>
-The second method is "git revert". Using "git revert" will create a new commit that is the same as the one you wish to revert to. Assume we've made changes on top of the HEAD node that we don't like anymore. Using "git revert HEAD" will create a new commit that is the same as the original version of HEAD.
+Finally, one last thing! There is a shortcut to reverting consecutive commits! For example, suppose we commit five files separately and we want to remove the changes made in commits 2 through 4 (i.e. remove files 2, 3, and 4). Use this syntax: "git revert <older_ref>..<newer_ref>". The older ref is non-inclusive; therefore, it should be the parent of the oldest commit to be reverted. The newer ref is inclusive; meaning it is the newest commmit to be reverted. All commits in-between will also be reverted. For our example: "git revert HEAD~4..HEAD^". This will revert the child of HEAD~4 (= C2) through HEAD^ (=C4).
 >>>
-Now, that seems redundant doesn't it? However, that's precisely why revert is important! By being redundant, we are keeping track of the changes we made in the edited version of HEAD. Therefore, if we decided we actually needed those changes, we can return to them. This is unlike reset, which modifies the commit history (you wouldn't be able to recover the reset data).
->>>
-Wow! You learned a lot (hopefully), prove yourself by completing the challenge!
+Wow! You learned a lot (hopefully), prove yourself by completing the challenge! Reverse all changes on the "local" branch and reverse changes made in commits 3 and 5 on the "remote" branch. Assume that the "local" and "remote" branches are as their names imply.
```

Comment:
You should tell them to remove commits 3 and 5 "in that order" since only one of the orderings is going to be correct.

Reply:
Just looked at `git revert 1..5` Since that  reverts 5 then 3, we should specify 5 then 3 as the order. It would also make sense to specify that `git revert x..y` works backwards somewhere in the instructions

Reply:
Oh, I want them to use the sequential revert

Reply:
So, you're thinking 
```
git revert 5
git revert 3
```
?

Reply:
I want them to use git revert HEAD~3..HEAD^2.

Reply:
We don't have any way of preventing them from doing things different ways right now, but that is the same net effect

Reply:
I specify to do it using one command in the instructions, the only way they could do it with one command is using the above command which forces that order. That being said, it's true that we can't prevent them from doing it using two commands. Although at the same time, if they did it with two commands and got the order wrong, they would think it's because they didn't do it with one command->then they would do it with one command and get it right :P